community/overrides/user.py

```diff
@@ -15,36 +15,33 @@ def validate(self):
     def validate_username_characters(self):
         if self.is_new():
 
+            if not self.username:
+                self.username = self.get_username_from_first_name()
+
             if self.username.find(" "):
                 self.username.replace(" ", "")
 
             if not re.match("^[A-Za-z0-9_]*$", self.username):
                 self.username = self.remove_illegal_characters()
 
-            if not self.username:
-                self.username = self.get_username_from_first_name()
-
-            if self.username_exists():
+            while self.username_exists():
                 self.username = self.remove_illegal_characters() + str(random.randint(0, 99))
```

Comment:
Interesting bug here. It is possible to make the system not accept signups with any given name. If I create 101 accounts with firstname "Foo", then the the system will go into infinite loop for the next person with first name "Foo".

Reply:
What if first name of a user is a single letter? Some people use abbreviations for the first name. The username generated will be an invalid username. It is good to make sure that the system will never generate an invalid username.